gto-support-kotlin-coroutines/src/main/kotlin/org/ccci/gto/android/common/kotlin/coroutines/ReadWriteMutex.kt

```diff
@@ -19,23 +19,22 @@ fun ReadWriteMutex(): ReadWriteMutex = ReadWriteMutexImpl()
 
 @VisibleForTesting
 internal class ReadWriteMutexImpl : ReadWriteMutex {
-    private val stateMutex = Mutex()
-    private val readerOwner = Any()
```

Comment:
the `readerOwner` wasn't really gaining us anything, so I removed it to prevent the race condition from triggering an exception.

The threads can still race, but now `write.lock()` will suspend until `write.unlock()` completes